ladybug_comfort/collection/solarcal.py

```diff
@@ -0,0 +1,530 @@
+        self._sky_exp = self._fraction_input_check(
+            sky_exposure, 'sky_exposure', 1)
+        self._flr_ref = self._fraction_input_check(
+            floor_reflectance, 'floor_reflectance', 0.25)
+        self._win_trans = self._fraction_input_check(
+            window_transmittance, 'window_transmittance', 0.4)
+
+        # check that all input data collections are aligned.
+        HourlyDiscontinuousCollection.are_collections_aligned(self._input_collections)
+
+        # check comfort parameters
+        self._body_par_check(solarcal_body_parameter)
+
+        # compute SolarCal
+        self._calculate_solarcal()
+
+    def _calculate_solarcal(self):
+        """Compute SolarCal for each step of the Data Collection."""
+        # empty lists to be filled
+        self._erf = []
+        self._dmrt = []
+        self._mrt = []
+
+        # get altitudes and sharps from solar position
+        _altitudes, _sharps = self._get_altitudes_and_sharps()
+
+        # calculate final erfs and mrt deltas
+        for l_mrt, diff, dir, alt, sharp, sky_e, fract_e, flr_ref, w_trans in \
+                zip(self._l_mrt, self._diff_horiz, self._dir_norm, _altitudes, _sharps,
+                    self._sky_exp, self._fract_exp, self._flr_ref, self._win_trans):
+
+            result = indoor_sky_heat_exch(l_mrt, diff, dir, alt, sky_e, fract_e,
+                                          flr_ref, w_trans, self._body_par.posture,
+                                          sharp, self._body_par.body_absorptivity,
+                                          self._body_par.body_emissivity)
+            self._erf.append(result['erf'])
+            self._dmrt.append(result['dmrt'])
+            self._mrt.append(result['mrt'])
+
+    @property
+    def diffuse_horizontal_solar(self):
+        """Data Collection of diffuse horizontal irradiance in Wh/m2 or W/m2."""
+        return self._diff_horiz.duplicate()
+
+    @property
+    def direct_normal_solar(self):
+        """Data Collection of direct normal irradiance in Wh/m2 or W/m2."""
+        return self._dir_norm.duplicate()
+
+    @property
+    def longwave_mrt(self):
+        """Data Collection of surface temperature values in degrees C."""
+        return self._l_mrt.duplicate()
+
+    @property
+    def sky_exposure(self):
+        """Data Collection of sky view."""
+        return self._sky_exp.duplicate()
+
+    @property
+    def window_transmittance(self):
+        """Data Collection of window transmittance."""
+        return self._win_trans.duplicate()
+
+    @property
+    def effective_radiant_field(self):
+        """Data Collection of shortwave effective radiant field in W/m2."""
+        return self._build_coll(self._erf, EffectiveRadiantField(), 'W/m2')
+
+    @property
+    def mrt_delta(self):
+        """Data Collection of shortwave MRT delta in C."""
+        return self._build_coll(self._dmrt, RadiantTemperatureDelta(), 'C')
+
+
+class HorizontalSolarCal(_SolarCalBase):
```

Comment:
Yes, these classes are basically wrappers around individual functions.  This one is around this function:
https://github.com/chriswmackey/ladybug-comfort/blob/master/ladybug_comfort/solarcal.py#L177

All that this wrapper is doing is making it easier to use these functions with data collections, location objects, and parameter objects.  Otherwise, you are basically leaving people with a function that has >10 inputs with required inputs that are not easy to obtain.